src/site/stages/build/plugins/modify-dom/process-entry-names.js

```diff
@@ -67,54 +67,48 @@ function getEntryNamesDictionary(buildOptions, files) {
-  return (files, metalsmith, done) => {
-    const entryNamesDictionary = getEntryNamesDictionary(buildOptions, files);
-
-    for (const fileName of Object.keys(files)) {
-      const file = files[fileName];
-
-      const { dom } = file;
-      if (!dom) continue;
-
-      dom('script[data-entry-name],link[data-entry-name]').each((index, el) => {
-        // Derive the element properties.
-        const $el = dom(el);
-        const entryName = $el.data('entryName');
-        const attribute = $el.is('script') ? 'src' : 'href';
-
-        // Derive the hashed entry name.
-        const hashedEntryName = entryNamesDictionary.get(entryName) || [];
-
-        // Assemble the filename so we can match it in the generated files array.
-        const fileSearch = `generated/${
-          hashedEntryName.split('/generated/')[1]
-        }`;
-
-        // Ensure we have valid options and that the entry exists.
-        const entryExists = files[fileSearch];
-
-        if (
-          buildOptions.buildtype !== environments.LOCALHOST &&
-          !buildOptions.isPreviewServer &&
-          !buildOptions.entry &&
-          !entryExists
-        ) {
-          throw new Error(`Entry Name "${entryName}" was not found.`);
-        }
-
-        // Link the element to the hashed entry name w/o the S3 bucket
-        $el.attr(attribute, `/${fileSearch}`);
-        file.modified = true;
-      });
-    }
+module.exports = {
+  initialize(buildOptions, files) {
+    this.entryNamesDictionary = getEntryNamesDictionary(buildOptions, files);
+  },
+
+  modifyFile(fileName, file, files, buildOptions) {
+    const { dom } = file;
+    if (!dom) return;
+
+    dom('script[data-entry-name],link[data-entry-name]').each((index, el) => {
+      // Derive the element properties.
+      const $el = dom(el);
+      const entryName = $el.data('entryName');
+      const attribute = $el.is('script') ? 'src' : 'href';
+
+      // Derive the hashed entry name.
+      const hashedEntryName = this.entryNamesDictionary.get(entryName) || [];
+
+      // Assemble the filename so we can match it in the generated files array.
+      const fileSearch = `generated/${hashedEntryName.split('/generated/')[1]}`;
+
+      // Ensure we have valid options and that the entry exists.
+      const entryExists = files[fileSearch];
+
+      if (
+        buildOptions.buildtype !== environments.LOCALHOST &&
+        !buildOptions.isPreviewServer &&
+        !buildOptions.entry &&
+        !entryExists
+      ) {
+        throw new Error(`Entry Name "${entryName}" was not found.`);
+      }
 
+      // Link the element to the hashed entry name w/o the S3 bucket
+      $el.attr(attribute, `/${fileSearch}`);
+      file.modified = true;
+    });
+  },
+
+  conclude(buildOptions, files) {
     if (!buildOptions.isPreviewServer) {
-      copyAssetsToTeamSitePaths(buildOptions, files, entryNamesDictionary);
+      copyAssetsToTeamSitePaths(buildOptions, files, this.entryNamesDictionary);
```

Comment:
I'm not certain this `copyAssetsToTeamSitePaths` function really does anything anymore. It seems like whatever is going on here has replaced it, https://github.com/department-of-veterans-affairs/vets-website/blob/b46b14fea6f04cedf7359d17e6c4d388400da0a2/config/webpack.config.js#L25.

Reply:
cc @timwright12 @cvalarida